std/path.d

```diff
@@ -2438,111 +2439,139 @@ in
 }
 body
 {
-    size_t ni; // current character in path
+    alias RC = Unqual!(ElementEncodingType!Range);
 
-    foreach (ref pi; 0 .. pattern.length)
+    static if (RC.sizeof == 1 && isSomeString!Range)
```

Comment:
Why the `isSomeString!Range`, you already check `isSomeChar!(ElementEncodingType!Range)`.